node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -56,6 +59,14 @@ pub struct Queues {
 
 	/// Priority queue.
 	priority: BTreeMap<CandidateComparator, ParticipationRequest>,
+
+	/// Timer handle for each participation request. Stored to measure full request
+	/// completion time. Optimally these would have been stored in the participation
+	/// request itself, but HistogramTimer doesn't implement the Clone trait.
+	request_timers: BTreeMap<CandidateComparator, Option<prometheus::HistogramTimer>>,
```

Comment:
> HistogramTimer doesn't implement the Clone trait.

I think you can put it behind a smart pointer like `Rc`. Might simplify the code. 🙂

Reply:
`Rc` doesn't work because we have multithreaded tokio runtime, we would need `Arc`.